lib/rugged_adapter/git_layer_rugged.rb

```diff
@@ -689,6 +685,16 @@ def find_branch(search_list)
 
     class Tree
 
+      def self.tree_entry_from_rugged_hash(entry, root = '')
+        {
+          sha:  entry[:oid],
+          mode: entry[:filemode].to_s(8),
+          type: entry[:type].to_s,
+          name: "#{root}#{entry[:name]}",
+          path: "#{root}#{entry[:path]}", # Repo#lstree is expected to give back an object containing :path rather than :name
```

Comment:
This is a bit unfortunate. `Repo#lstree` is called from gollum lib, and there it expects the resulting hash to have `:path` instead of `:name`. Should we rename this in `gollum-lib`, or is this way of dealing with it (having both `:name` and `:path` in this hash) acceptable?